kubernetes-client/src/main/java/io/fabric8/kubernetes/client/internal/CertUtils.java

```diff
@@ -145,7 +145,7 @@ private static PrivateKey handleECKey(InputStream keyInputStream) throws IOExcep
         @Override
         public PrivateKey call() {
           try {
-            if (Security.getProvider("BC") == null) {
+            if (Security.getProvider("BC") == null && Security.getProvider("BCFIPS") == null) {
               Security.addProvider(new org.bouncycastle.jce.provider.BouncyCastleProvider());
```

Comment:
Shouldn't you be adding the `BouncyCastleFipsProvider` here if it's not found, instead of the regular BC provider?

Reply:
According to [Bouncy Castle FIPS Java API User Guide](https://downloads.bouncycastle.org/fips-java/BC-FJA-UserGuide-1.0.2.pdf) security provider can be added in **java.security** file.

I chose to support only the external configuration due to the following reasons:
- Additional flag for FIPS mode isn't required
- No need to add dependency with BC FIPS and update versions
- Adding FIPS provider at the state of initialization of Kubernetes client is too late. So much code can be executed in an application using kubernetes-client. Initialization on startup of JVM is preferable.

Reply:
As I understand, this statement prevents the the BouncyCastleProvider to be added if you configured your JVM to use the BCFIPS provider, right?

Reply:
On the other hand, these changes should be harmless in case no BC configuration is provided, right?

Reply:
Hmm, still not convinced… Doesn't this mean that someone might be requesting the FIPS provider and actually getting the regular one?

Reply:
@manusa , yes! Changes should be harmless in case no BC configuration is provided.
There are 4 cases:
1. None of the BouncyCastle security providers configured in JVM -> BC will be added in runtime
2. BC configured in JVM -> BC will not be added in runtime
3. BCFIPS configured in JVM -> BC will not be added in runtime
4. Another FIPS provider configured in JVM (is it exist?) -> BC will be added in runtime

@metacosm , looks like you are asking about case №4, aren't you?
I see 2 solutions: 
1. The additional flag can be added to prevent adding BC provider in case if someone would like to use an external provider.
2. We can say that kubernetes-client works only with the family of BouncyCastle security providers (BC/BCFIPS).

Reply:
What I don't understand is why we add the BC provider when the user asks for `BCFIPS` and not the `BouncyCastleFipsProvider`? How will that address the issue?

Reply:
@metacosm , because there is no option in kubernetes-client configuration for add `BouncyCastleFipsProvider(BCFIPS)` instead of `BouncyCastleProvider(BC)`.

But if the user will configure JVM for use `BouncyCastleFipsProvider(BCFIPS)` kubernetes-client will not add `BouncyCastleProvider(BC)`. See case №3 in my previous message.